components/icon/src/build.js

```diff
@@ -1,94 +1,97 @@
-  if (err) throw err;
-}
-
-const directoryPath = path.join(__dirname, '../tmp/optimized-svgs');
-
-fs.readdir(directoryPath, function (err, files) {
-  //handling error
-  if (err) {
-    console.error('Unable to scan directory: ' + err);
-    return;
+const cwd = path.resolve(process.cwd(), 'tmp/optimized-svgs');
+
+const readSVGDir = async () => {
+  let files;
+  try {
+    files = await readdir(cwd);
+    return files;
+  } catch (error) {
+    console.error(error);
   }
+  return files;
+};
 
-  const iconData = files.map((file) => ({
-    id: `${componentPrefix}${kebabCase(file.replace('.svg', ''))}`,
-    src: file,
-  }));
-
-  files.forEach(function (file) {
-    const fileName = `${componentPrefix}${kebabCase(file.replace('.svg', ''))}`;
-    const formattedName = kebabCase(fileName);
-    const dir = `${componentsPath}/${formattedName}`;
-
-    fs.readFile(path.join(directoryPath, file), 'utf8', function (err, svg) {
-      if (err) {
-        console.error(err);
-        return;
-      }
-
-      // throw an error if the file already exists
-      if (fs.existsSync(dir)) throw new Error('A component with that name already exists.');
-
-      // create the folder
-      fs.mkdirSync(dir);
-      fs.writeFile(
-        `${dir}/${formattedName}.stencil.tsx`,
-        component(fileName, svg, iconRapperComponent),
-        writeFileErrorHandler,
+const getIconData = async () => {
+  let svgData;
+
+  const files = await readSVGDir();
+  try {
+    if (files && files.length > 0) {
+      svgData = await Promise.all(
+        files.map((file) => ({
+          id: `${componentPrefix}${kebabCase(path.parse(file).name)}`,
+          src: file,
+        })),
       );
-      fs.writeFile(`${dir}/${formattedName}.spec.tsx`, test(fileName), writeFileErrorHandler);
-    });
-  });
-
-  const dirExistsSync = (dir) => {
-    const stats = !!fs.statSync(dir, { throwIfNoEntry: false });
-    return stats ? stats.isDirectory() : false;
-  };
-
-  const outputDir = path.resolve(__dirname, '../dist/');
+    }
+  } catch (error) {
+    console.error(error);
+  }
+  return svgData;
+};
 
-  if (!dirExistsSync(outputDir)) {
-    fs.mkdirSync(outputDir);
+const WriteComponentFile = async () => {
+  try {
+    const files = await readSVGDir();
+    if (files && files.length > 0) {
+      for (const file of files) {
+        const fileBasename = path.parse(file).name;
+        const fileName = `${componentPrefix}${kebabCase(fileBasename)}`;
+        const formattedName = kebabCase(fileName);
+        const svg = await readFile(`${cwd}/${file}`, 'utf-8');
+
+        await mkdir(`${componentsPath}/${formattedName}`, { recursive: true });
+        await writeFile(
+          `${componentsPath}/${formattedName}/${formattedName}.stencil.tsx`,
+          component(fileName, svg, iconRapperComponent),
+          'utf-8',
+        );
+        await writeFile(`${componentsPath}/${formattedName}/${formattedName}.spec.tsx`, test(fileName), 'utf-8');
```

Comment:
In the future you could consider an approach where you don't `await` for every file to finish writing, but to start writing to a couple of files in parallel and just `await Promise.all(...)`